src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -348,107 +440,120 @@ internal enum BuildProgress
 
         private sealed class InProgressState
```

Comment:
can you follow this up with a PR that breaks these into separate files.  there's just a ton of complexity in these files.

Reply:
Yes, sure.